Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -337,29 +332,27 @@ extension AWSClient {
     ///     - url : URL to sign
     ///     - httpMethod: HTTP method to use ("GET", "PUT", "PUSH" etc)
     ///     - expires: How long before the signed URL expires
+    ///     - serviceConfig: Additional configuration to sign the url
     /// - returns:
     ///     A signed URL
-    public func signURL(url: URL, httpMethod: String, expires: Int = 86400) -> EventLoopFuture<URL> {
-        return signer.map { signer in signer.signURL(url: url, method: HTTPMethod(rawValue: httpMethod), expires: expires) }
+    public func signURL(url: URL, httpMethod: String, expires: Int = 86400, serviceConfig: AWSServiceConfig) -> EventLoopFuture<URL> {
```

Comment:
Let's stick with serviceConfig here. The signing name is not always obvious for each service. 
```
return client.signURL(url: myURL, httpMethod: "GET", serviceConfig: S3.serviceConfig)
```